orderbook/src/orderbook.rs

```diff
@@ -109,7 +109,10 @@ impl Orderbook {
             None => return Ok(OrderCancellationResult::OrderNotFound),
         };
 
-        match cancellation.validate_signature(&self.domain_separator) {
+        match cancellation
```

Comment:
nanonit: can we keep the `validate_signature` method on the cancellation struct? I find this nicer to interact with than having the orderbook know how cancellations are validated.

Reply:
fixed.